aptos/contract/aux/sources/stake.move

```diff
@@ -1517,4 +1575,109 @@ module aux::stake {
             assert!(user_pos.last_acc_reward_per_share == pool.acc_reward_per_share, E_TEST_FAILURE);
         };
     }
+
+    #[test(sender = @0x5e7c3, aptos_framework = @0x1, alice = @0x123)]
+    fun test_end_reward_early(sender: &signer, aptos_framework: &signer, alice: &signer) acquires Pools, UserPositions {
+        setup_module_for_test(sender, aptos_framework);
+        let sender_addr = signer::address_of(sender);
+        let alice_addr = signer::address_of(alice);
+        if (!account::exists_at(alice_addr)) {
+            account::create_account_for_test(alice_addr);
+        };
+
+        let sender_eth = 5 * 100000000;
+        let alice_eth = sender_eth;
+        fake_coin::register_and_mint<USDC>(sender, 3000000 * 1000000); // 2M USDC
+        fake_coin::register_and_mint<ETH>(sender, sender_eth); // 5 ETH
+        fake_coin::register_and_mint<USDC>(alice, 0); // 2M USDC
+        fake_coin::register_and_mint<ETH>(alice, alice_eth); // 5 ETH
+        let reward_au = 2000000 * 1000000;
+        let reward_remaining = reward_au;
+        let reward = coin::withdraw<FakeCoin<USDC>>(sender, reward_au);
+
+        // Incentive:
+        // duration = 30 days = 30*24*3600*1000000 microseconds
+        // reward = 2e14 AU ETH
+        // reward per second = 77,160,493.82716049
+        let duration_seconds = 30*24*3600; // 30 days
+        let start_time = timestamp::now_microseconds();
+        let end_time = start_time + duration_seconds * 1000000;
+        let pool_id = create<FakeCoin<ETH>, FakeCoin<USDC>>(sender_addr, reward, end_time);
+        let sender_usdc = 1000000 * 1000000;
+        assert!(coin::balance<FakeCoin<USDC>>(sender_addr) == sender_usdc, E_TEST_FAILURE);
+        {
+            let pools = borrow_global_mut<Pools<FakeCoin<ETH>, FakeCoin<USDC>>>(@aux);
+            let pool = table::borrow_mut(&mut pools.pools, pool_id);
+            assert!(pool.authority == sender_addr, E_TEST_FAILURE);
+            assert!(pool.start_time == start_time, E_TEST_FAILURE);
+            assert!(pool.end_time == end_time, E_TEST_FAILURE);
```

Comment:
Not so sure how this test was passingwith seemingly stale input parameters for create function line 1605. Maybe I'm missing something...

Reply:
Ah, that's a good catch. I think it's because in the tests, the timestamp starts as `0`, so there's no difference between `end_time` and `duration_us`. I'll fast forward time a bit in the setup to avoid that.

Reply:
Alright I added the fix and rebased, but if you check out this commit: `0f2761d65443b06ea73f409969f2d1fd82b29586` you should see these tests failing due to that bug.

Reply:
Thanks, Leina! I cannot replicate the test failures : (
`
misung@penguin:~/aux-exchange/aptos/api/aux-ts$ git checkout 0f2761d65443b06ea73f409969f2d1fd82b29586
fatal: reference is not a tree: 0f2761d65443b06ea73f409969f2d1fd82b29586
`
I was able to check out another commit https://github.com/aux-exchange/aux-exchange/pull/296/commits/f854bb8279cec11c433a3272981d3890f838caf1 instead, but I'm not so sure how to run these tests in `stake.move` instead of `stake.spec.ts`:
`
misung@penguin:~/aux-exchange/aptos/api/aux-ts$ git checkout f854bb8279cec11c433a3272981d3890f838caf1
HEAD is now at f854bb8 fast forward start time in stake tests
misung@penguin:~/aux-exchange/aptos/api/aux-ts$ APTOS_NETWORK="local" yarn test --g "Stake"


  Stake Pool tests
    ✔ fundAccount (1113ms)
    ✔ createStakePoolClient
    ✔ mintAux (2482ms)
    ✔ mintOther (1073ms)
    ✔ createStakePool (1069ms)
    ✔ deposit (1153ms)
Sleeping for 2 seconds
    ✔ claim (3057ms)
    ✔ modifyPool (1065ms)
    ✔ modifyAuthority (3305ms)
    ✔ endRewardEarly (2191ms)
    ✔ withdraw (1128ms)
    ✔ deleteEmptyPool (1075ms)


  12 passing (19s)
`